placeSpecificContent/hq/chatterboxes/chatterbox/buttons/privacyButton_server.js

```diff
@@ -0,0 +1,396 @@
+            } else {
+                _this.playSound(DONE_SOUND, DONE_VOLUME, buttonPosition, false, false);
+                currentState = (currentState === "closed" || currentState === "toClosed") ? "open" : "closed";
+                Entities.editEntity(button, {
+                    color: currentState === "closed" ? BUTTON_CLOSED_COLOR : BUTTON_OPEN_COLOR
+                });
+                return;
+            }
+            
+            _this.playSound(TRANSITION_SOUND, TRANSITION_VOLUME, buttonPosition, false, true);
+
+            // Start door movement in whichever axis it should move along first
+            if (doorMovements.xVelocity && doorMovements.currentAxis === "x") {
+                Entities.editEntity(door, {
+                    localVelocity:  { x: doorMovements.xVelocity, y: 0, z: 0 }
+                });
+            } else if (doorMovements.zVelocity && doorMovements.currentAxis === "z") {
+                Entities.editEntity(door, {
+                    localVelocity:  { x: 0, y: 0, z: doorMovements.zVelocity }
+                });
+            }
+            
+            // Make sure the shade is visible if beginning to close it
+            if (currentState === "toClosed") {
+                Entities.editEntity(shade, {
+                    visible: true
+                });
+            }
+
+            updateInterval = Script.setInterval(function() {
+                _this.updateEntities(doorMovements, shadeTargetHeight);
+            }, UPDATE_INTERVAL_MS);
+        },
+
+        /* Handle any odd scenarios where the door is not nearby or is already in place and then get data for how it will move*/
+        setUpDoorMovements: function(doorMovements) {
+            var doorCurrentLocalPosition = Entities.getEntityProperties(door, 'localPosition').localPosition;
+            var doorDistanceToTarget = Vec3.distance(doorCurrentLocalPosition, doorMovements.targetPosition);
+            // If the door has somehow been moved far away, slam it into open position to avoid hitting anyone in the doorway
+            if (doorDistanceToTarget > MAX_DISTANCE_TO_TRANSITION_ENTITIES_M) {
+                Entities.editEntity(door, {
+                    localPosition: DEFAULT_DOOR_OPEN_LOCAL_POSITION
+                });
+            // If door is not aligned vertically, slam into correct vertical position
+            } else if (doorCurrentLocalPosition.y !== DEFAULT_DOOR_CLOSED_LOCAL_POSITION.y) {
+                doorCurrentLocalPosition = {
+                    x: doorCurrentLocalPosition.x,
+                    y: DEFAULT_DOOR_CLOSED_LOCAL_POSITION.y,
+                    z: doorCurrentLocalPosition.z
+                };
+                Entities.editEntity(door, {
+                    localPosition: doorCurrentLocalPosition
+                });
+            // If door is very close to correct position, slam into exact position
+            } else if (doorDistanceToTarget <= MIN_DISTANCE_TO_TRANSITION_ENTITIES_M) {
```

Comment:
why is this an `else if` instead of just a separate `if` ?

Reply:
It should be an else if but after the check for if the door is too far away and then check if the door is aligned vertically. So, if the door is too far, else if the door is too close to the target position, else if the door is not vertically aligned...